src/web/src/hooks/use-telescope-contributor.ts

```diff
@@ -12,31 +11,80 @@ export type Contributor = {
 const telescopeGitHubContributorsUrl =
   'https://api.github.com/repos/Seneca-CDOT/telescope/contributors';
 
-const randomContributor = (contributors?: any[]) =>
-  contributors ? contributors[Math.floor(Math.random() * contributors.length)] : null;
+const prepareURL = (page?: number) =>
+  `${telescopeGitHubContributorsUrl}?per_page=1&${page ? `page=${page}` : ''}`;
 
-const PER_PAGE: number = 50;
+const linkParser = (links: string, key: string) => {
+  let url;
+  let page;
+  const regex = new RegExp(`<([^?]+\\?per_page=1&[a-z]+=([\\d]+))>;[\\s]*rel="${key}"`, 'g');
+  const arr = regex.exec(links);
 
-const useTelescopeContributor = () => {
-  const [contributor, setContributor] = useState<Contributor | null>(null);
+  if (arr !== null) {
+    [, url, page] = arr;
+  }
+
+  return { url, page };
+};
+
+const getMaxContributors = async () => {
+  const response = await fetch(prepareURL());
+  const links = response.headers.get('Link') || '';
+
+  // Extract page count from response headers
+  const { page: max_contributors } = linkParser(links, 'last');
+  if (window.localStorage) window.localStorage.setItem('fetch_date', String(Date.now()));
+
+  return max_contributors;
+};
+
+const getContributor = async (randomPage: number) => {
+  const response = await fetch(prepareURL(randomPage));
```

Comment:
Various things can fail in this function, you need to try/catch either here or in the caller and not let it crash the frontend if something fails (GitHub down, 429 response, data comes back not in array form, empty array, etc)